src/serializers/type_serializers/json_or_python.rs

```diff
@@ -59,7 +59,7 @@ impl TypeSerializer for JsonOrPythonSerializer {
     }
 
     fn json_key<'py>(&self, key: &'py PyAny, extra: &Extra) -> PyResult<Cow<'py, str>> {
-        self._invalid_as_json_key(key, extra, Self::EXPECTED_TYPE)
+        self.json.json_key(key, extra)
```

Comment:
@adriangb looks like you wrote the original implementation in #598, something I'm missing?

Reply:
Probably not. I expect I did this as an "I'm not sure what should happen here, let's just be safe" sort of move.